elements/lisk-p2p/test/integration/p2p.ts

```diff
@@ -1030,6 +976,193 @@ describe('Integration tests for P2P library', () => {
+					p2p.on('messageReceived', message => {
+						collectedMessages.push({
+							nodePort: p2p.nodeInfo.wsPort,
+							message,
+						});
+						let peerRates = messageRates.get(p2p.nodeInfo.wsPort) || [];
+						peerRates.push(message.rate);
+						messageRates.set(p2p.nodeInfo.wsPort, peerRates);
+					});
+				}
+			});
+
+			it('should track the message rate correctly when receiving messages', async () => {
+				const TOTAL_SENDS = 100;
+				const firstP2PNode = p2pNodeList[0];
+				const ratePerSecondLowerBound = 70;
+				const ratePerSecondUpperBound = 130;
+
+				const targetPeerPort = NETWORK_START_PORT + 3;
+				const targetPeerId = `127.0.0.1:${targetPeerPort}`;
+
+				for (let i = 0; i < TOTAL_SENDS; i++) {
+					await wait(10);
+					firstP2PNode.sendToPeer(
+						{
+							event: 'foo',
+							data: i,
+						},
+						targetPeerId,
+					);
+				}
+
+				await wait(50);
+				const secondPeerRates = messageRates.get(targetPeerPort) || [];
+				const lastRate = secondPeerRates[secondPeerRates.length - 1];
+
+				expect(lastRate).to.be.gt(ratePerSecondLowerBound);
+				expect(lastRate).to.be.lt(ratePerSecondUpperBound);
+			});
+
+			it('should disconnect the peer if it tries to send messages at a rate above wsMaxMessageRate', async () => {
+				const TOTAL_SENDS = 300;
+				const firstP2PNode = p2pNodeList[0];
+				const secondP2PNode = p2pNodeList[1];
+
+				const removedPeers: Array<string> = [];
+
+				secondP2PNode.on(EVENT_REMOVE_PEER, peerId => {
+					removedPeers.push(peerId);
+				});
+
+				const targetPeerId = `127.0.0.1:${secondP2PNode.nodeInfo.wsPort}`;
+				const failedToSendErrors: Array<Error> = [];
+
+				for (let i = 0; i < TOTAL_SENDS; i++) {
+					await wait(1);
+					try {
+						firstP2PNode.sendToPeer(
+							{
+								event: 'foo',
+								data: i,
+							},
+							targetPeerId,
+						);
+					} catch (error) {
+						failedToSendErrors.push(error);
+					}
+				}
+
+				await wait(100);
+
+				expect(removedPeers).to.contain('127.0.0.1:5000');
+			});
+		});
+
+		describe('P2P.requestFromPeer', () => {
+			let collectedMessages: Array<any> = [];
+			let requestRates: Map<number, Array<number>> = new Map();
+
+			beforeEach(() => {
+				collectedMessages = [];
+				requestRates = new Map();
+				for (let p2p of p2pNodeList) {
+					p2p.on('requestReceived', request => {
+						collectedMessages.push({
+							nodePort: p2p.nodeInfo.wsPort,
+							request,
+						});
+						if (request.procedure === 'getGreeting') {
+							request.end(
+								`Hello ${request.data} from peer ${p2p.nodeInfo.wsPort}`,
+							);
+						} else {
+							if (!request.wasResponseSent) {
+								request.end(456);
+							}
+						}
+						let peerRates = requestRates.get(p2p.nodeInfo.wsPort) || [];
+						peerRates.push(request.rate);
+						requestRates.set(p2p.nodeInfo.wsPort, peerRates);
+					});
+				}
+			});
+
+			it('should track the request rate correctly when receiving requests', async () => {
```

Comment:
Shall we add another test case to check if the peer interrupts the channel?

Reply:
@diego-G Ok resolved.